src/System.CommandLine.ApiCompatibility.Tests/ApiCompatibilityApprovalTests.System_CommandLine_api_is_not_changed.approved.txt

```diff
@@ -217,14 +212,27 @@ System.CommandLine.Help
   public class HelpOption : System.CommandLine.CliOption<System.Boolean>
     .ctor()
     .ctor(System.String name, System.String[] aliases)
-    public System.CommandLine.CliAction Action { get; set; }
+    public System.CommandLine.Invocation.CliAction Action { get; set; }
   public class TwoColumnHelpRow, System.IEquatable<TwoColumnHelpRow>
     .ctor(System.String firstColumnText, System.String secondColumnText)
     public System.String FirstColumnText { get; }
     public System.String SecondColumnText { get; }
     public System.Boolean Equals(System.Object obj)
     public System.Boolean Equals(TwoColumnHelpRow other)
     public System.Int32 GetHashCode()
+System.CommandLine.Invocation
+  public abstract class AsynchronousCliAction : CliAction
+    public System.Threading.Tasks.Task<System.Int32> InvokeAsync(System.CommandLine.ParseResult parseResult, System.Threading.CancellationToken cancellationToken = null)
+  public abstract class CliAction
+    public System.Boolean Terminating { get; }
+    protected System.Void set_Terminating(System.Boolean value)
+  public class ParseErrorAction : SynchronousCliAction
```

Comment:
My thought here was that since the entire invocation space is a effectively a different layer and could be replaced with a completely different invocation infrastructure on top of the same parser, none of these belong under parsing.